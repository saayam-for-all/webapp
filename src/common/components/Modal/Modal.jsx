import React from "react";

const Modal = ({ show, onClose, children }) => {
  if (!show) return null;

  return (
    <div className="fixed inset-0 bg-gray-600 bg-opacity-50 flex justify-center items-center z-50">
      <div className="bg-white p-6 rounded-lg shadow-lg max-w-2xl w-full">
        {children}
        <button
          onClick={onClose}
          className="mt-4 bg-blue-500 text-white px-4 py-2 rounded-lg hover:bg-blue-500"
        >
          Close
        </button>
      </div>
    </div>
  );
};

export default Modal;
