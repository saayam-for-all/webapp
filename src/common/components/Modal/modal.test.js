import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import Modal from "./Modal";

const children = <div>child</div>;
const onClose = jest.fn();

describe("Modal", () => {
  it("renders when show = true", () => {
    const tree = render(
      <Modal show={true} onClose={onClose}>
        {children}
      </Modal>,
    );
    expect(tree).toMatchSnapshot();
  });

  it("does not render when show = false", () => {
    const tree = render(
      <Modal show={false} onClose={onClose}>
        {children}
      </Modal>,
    );
    expect(tree.container).toBeEmptyDOMElement();
  });
});
