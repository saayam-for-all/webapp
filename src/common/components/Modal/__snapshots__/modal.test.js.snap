// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders when show = true 1`] = `
<div>
  <div
    class="fixed inset-0 bg-gray-600 bg-opacity-50 flex justify-center items-center z-50"
  >
    <div
      class="bg-white p-6 rounded-lg shadow-lg max-w-2xl w-full"
    >
      <div>
        child
      </div>
      <div
        class="flex justify-end space-x-2 mt-6"
      >
        <button
          class="bg-gray-400 text-white px-3 py-1 text-sm rounded hover:bg-gray-500 transition"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
